Triangulation_2/include/CGAL/Triangulation_2_projection_traits_3.h

```diff
@@ -259,41 +302,69 @@ class Triangulation_2_projection_traits_3
 {
   typedef Triangulation_2_projection_traits_3<Kernel> Self;
 
-  typename Kernel::Vector_3 n;
+  typename Kernel::Vector_3 n, b1, b2;
 
 public:
   typedef typename Kernel::Vector_3 Vector_3;
 
 
-  Triangulation_2_projection_traits_3(const Vector_3& n_)
+  explicit Triangulation_2_projection_traits_3(const Vector_3& n_)
     : n(n_)
-  {}
+  {
+    typedef typename Kernel::FT FT;
+    typedef typename Kernel::Vector_3 Vector_3;
+
+    const FT& nx = n.x();
+    const FT& ny = n.y();
+    const FT& nz = n.z();
+    const FT anz = CGAL::abs(nz);
+    if(anz >= CGAL::abs(nx) &&
+       anz >= CGAL::abs(ny)) { // z is the max absolute coordinate of n
+      b1 = Vector_3(nz, 0, -nx);
+    }
+    else {
+      b1 = Vector_3(ny, -nx, 0);
+    }
+    b2 = cross_product(n, b1);
+    // std::cerr << "\nTriangulation_2_projection_traits_3(" << b1 << "\n"
+    // 	      << "                                    " << b2 << "\n"
+    // 	      << "                                    " << n << "\n";
+  }
 
   Triangulation_2_projection_traits_3(const Self& other)
-    : n(other.n)
+    : n(other.n), b1(other.b1), b2(other.b2)
   {
 //     std::cerr << "Copy of a traits. Type="
 //               << typeid(*this).name() << std::endl
 //               << "normal=" << normal() << std::endl;
   }
 
```

Comment:
No. It was used to add debugging traces, in the body of the copy-constructor. Do you think I should remove it before the integration? That copy-constructor is already in `master`.